.github/actions/vcpkg-cache-action/main.py

```diff
@@ -0,0 +1,197 @@
+import asyncio
+import contextlib
+import logging
+import os
+import signal
+import socket
+import subprocess
+import sys
+import tempfile
+
+import aiohttp
+import aiofiles
+
+LOCALHOST = "127.0.0.1"
+
+logging.basicConfig()
+logger = logging.getLogger()
+
+
+class ProxyStartError(RuntimeError):
+    pass
+
+
+def find_free_port() -> int:
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind((LOCALHOST, 0))
+        return s.getsockname()[1]
```

Comment:
I guess there's a very small chance of a race condition here if anything else was to try to bind to a port after this disconnects, compared to the approach of letting uvicorn pick a port and reading it from the log. I think this is fine though and a lot simpler.